File: core/java/src/main/java/hara/lang/protocol/IEquality.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iequality", name = "IEquality")
public interface IEquality {
  @HaraMethod(value = "equality", arity = 2)
  boolean equality(Object other);
}

File: core/rust/src/lang/protocol/iequality.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iequality", name = "IEquality")]
pub trait IEquality<Rhs: ?Sized = Self> {
    #[hara_method(value = "equality", arity = 2)]
    fn equality(&self, other: &Rhs) -> bool;
}
